api/c_appendix.txt

```diff
@@ -77,28 +77,28 @@ The application scalar and vector data types are defined in the
 The user is responsible for ensuring that pointers passed into and out of
 OpenCL kernels are natively aligned relative to the data type of the
 parameter as defined in the kernel language and SPIR-V specifications.
-This implies that OpenCL buffers created with CL_MEM_USE_HOST_PTR need to
+This implies that OpenCL buffers created with {CL_MEM_USE_HOST_PTR} need to
 provide an appropriately aligned host memory pointer that is aligned to the
 data types used to access these buffers in a kernel(s), that SVM allocations
 must correctly align and that pointers into SVM allocations must also be
 correctly aligned.
 The user is also responsible for ensuring image data passed is aligned to
 the granularity of the data representing a single pixel (e.g.
-`image_num_channels * sizeof(image_channel_data_type)`) except for CL_RGB
-and CL_RGBx images where the data must be aligned to the granularity of a
+`image_num_channels * sizeof(image_channel_data_type)`) except for {CL_RGB}
+and {CL_RGBx} images where the data must be aligned to the granularity of a
 single channel in a pixel (i.e. `sizeof(image_channel_data_type)`).
-This implies that OpenCL images created with CL_MEM_USE_HOST_PTR must align
+This implies that OpenCL images created with {CL_MEM_USE_HOST_PTR} must align
 correctly.
 The image alignment value can be queried using the
-CL_DEVICE_IMAGE_BASE_ADDRESS_ALIGNMENT query.
-In addition, source pointers for *clEnqueueWriteImage* and other operations
+{CL_DEVICE_IMAGE_BASE_ADDRESS_ALIGNMENT} query.
+In addition, source pointers for {clEnqueueWriteImage} and other operations
 that copy to the OpenCL runtime, as well as destination pointers for
-*clEnqueueReadImage* and other operations that copy from the OpenCL runtime
+{clEnqueueReadImage} and other operations that copy from the OpenCL runtime
 must follow the same alignment rules.
 
 OpenCL makes no requirement about the alignment of OpenCL application
 defined data types outside of buffers and images, except that the underlying
-vector primitives (e.g. `+__cl_float4+`) where defined shall be directly
+vector primitives (e.g. `__cl_float4`) where defined shall be directly
```

Comment:
I've verified that both the HTML and PDF outputs are generated correctly after removing the ++ delimiters, at least with my version of asciidoctor.

I'm fine putting them back though, if that's safer.  The docs hint that it's good practice, at least:

https://asciidoctor.org/docs/user-manual/#escaping-unconstrained-quotes

Reply:
What version are you using? Thanks to some backwards-incompatible changes in asciidoctor 2.0.x, I think Vulkan is going to be locked into 1.5.8, for a considerable time.

My recollection with the +/_ situation is that I tried to do it in just cases that absolutely required it, at first. Then after a couple of paragraphs got reflowed and that changed the _ handling, I got more aggressive. But it's been a while since I was dealing with that and if 1.5.8 is robust and stable in un-++-ed _ handling, yay!

Reply:
So this line renders correctly for me, but other examples that change `` `+__foo` `` to `` `__foo` `` don't render the `__` part of the text.

Observationally, ones that work (such as this one) only use `__` once in the paragraph, whereas the ones that don't use it twice.  It doesn't apply the formatting to all the text inbetween though.

This is with Asciidoctor 2.0.10, which I'm not sure whether it is expected to work (I just wiped/rebuilt my laptop, so only just started using this version).

Related, I've also just started a local branch because sometimes the string `C++` gets rendered as `C` (but again, not all always).

Reply:
I'll revert this change.  It's not required, and it seems like it breaks at least some versions of Asciidoctor.

Reply:
I've reverted this change.

Reply:
Asciidoctor 2.0 is not backwards-compatible with 1.x; they made choices that break some of the markup that we use in Vulkan (and probably use in CL as well, since I did that conversion). The differences probably aren't huge but for the time being, I think it's best to stay on 1.5.8 or thereabouts. This is not a hardship given the way ruby gems work.

Reply:
Specifically re the double-underscore markup, I put in all the ++ delimiters precisely because it was hard to understand the conditions under which __ would or would not be treated as markup - and whether it would or not was dependent not necc. on the __ you were looking at, but whether there was another one present within (some textual distance). I get that asciidoctor 2.0 is trying to improve stuff like this by actually having a semantic model of the markup, but in the process they broke *other* things we rely on :-(

Reply:
@oddhack I see. I'll install 1.5.8 then.